src/modules/hmis/components/HmisFilter.tsx

```diff
@@ -69,6 +69,7 @@ const FILTER_NAME_TO_PICK_LIST = {
   project: PickListType.Project,
   appliedToProject: PickListType.Project,
   organization: PickListType.Organization,
+  assessmentType: PickListType.AssessmentTypes,
```

Comment:
Hmm actually, I am overlooking the "pickListArgs" piece. The filter cannot infer the project id for filters, that needs to be passed explicitly (which you currently do with the `filters` prop). 

I don't think we have a clean way to do that. I'm going to post these review comments because I need to head about but I think it requires some further thinking on an API for filtering that works for dynamic pick lists with arguments. We can check in tomorrow!

Reply:
Thanks!! I sketched a possible solution here: https://github.com/greenriver/hmis-frontend/commit/8d21b0ac8e27c415242eab6587e2083b2f32c562, adding a new prop `filterArgs` (could be renamed to `filterPickListArgs`?) on `GenericTableWithData`. `filterArgs` maps filter names to their pick list args, and is passed all the way down to the `getFilterForType` function.

I would still need to update `type` to `assessmentType` as you said in your comment above. What do you think of this approach?

Reply:
That sounds reasonable @martha. I see that your approach allows you to pass different pick list args to different filters, for example, pass `projectId` to one pick list but not to another. I don't think we really need that, I can't think of a use case, but there may be one. My only other comment would be to add typing (use existing `PickListArgs`) and use the filterPickListArgs naming for clarity.

--

This is certainly scope creep from the original ticket, but, I think some light refactoring would help us here.. It feels wrong that the `GenericTableWithData` component is responsible for the logic of generating the filter definition object based on the filter type (eg `AssessmentsForProjectFilterOptions`). It's making it so we need to keep adding more filter-related props to the table component. That component is already responsible for a lot, so it feels like something that could be pulled out.


I'm thinking about something like the below. What do you think? This would be more of a developer ergonomics change that anything else. it would make it a bit simpler to implement the new pick list requirement imo.

```jsx
const MyTable () => {
   // returns memoized filter object `TableFilterType<FilterOptionsType>`
   // this hook would encapsulate the logic thats currently in the allFieldFilters fn
   const filters = useFilters({
      type: 'AssessmentsForProjectFilterOptions', // filter input type from schema
      omit: ['searchTerm'], // optional: skip some filters
      pickListArgs: { projectId }, // optional: pick list args to be applied to all PickList filter items
     // could add other args to hook as needed, without messing with GenericTable component
    })

   // filters can be modified as needed

   return (
     <GenericTableWithData...
        filters={filters} // filter object, that may have been constructed manually or generated from schema
        defaultFilters={ {status: Status.OPEN } } // default  filter *values* still accepted
        // filterInputType - deprecate. we could still keep it around and GenericTableWithData could use the useFilter hook internally, or, we could refactor all 33 usages
     />
   )
}
```

Reply:
Discussed and decided to attempt the refactor, timeboxed to ~2-3 hours. Result of my efforts here: https://github.com/greenriver/hmis-frontend/pull/749